rectools/models/pure_svd.py

```diff
@@ -93,6 +107,16 @@ def __init__(
         self.tol = tol
         self.maxiter = maxiter
         self.random_state = random_state
+        if use_gpu:  # pragma: no cover
+            if not cp:
+                warnings.warn("Forced to use CPU. CuPy is not available.")
+                use_gpu = False
+            elif not cp.cuda.is_available():
+                warnings.warn("Forced to use CPU. GPU is not available.")
+                use_gpu = False
+
+        self.use_gpu = use_gpu
+        self._use_gpu = use_gpu  # for making a config
```

Comment:
We need to do this before changing the given value of `use_gpu` since we want to keep in config the values that were used by user while creating the model

Reply:
Fixed b92ed6b